js/ai/src/generate.ts

```diff
@@ -629,6 +629,17 @@ export async function generate<
   );
 }
 
+function stripUndefinedOptions(input?: any): any {
+  if (!input) return input;
+  const copy = { ...input };
+  Object.keys(input).forEach((key) => {
+    if (copy[key] === undefined) {
+      delete copy[key];
+    }
+  });
+  return copy;
+}
+
```

Comment:
yeah, might be useful elsewhere, but it is "specifically built" for options... I'm going to keep it for now, it can be refactored to use another more general purpose helper in the future.